apps/prairielearn/src/pages/courseSyncs/courseSyncs.js

```diff
@@ -24,84 +25,63 @@ router.get(
       throw new HttpStatusError(403, 'Access denied (must be course editor)');
     }
 
-    const jobSequencesResult = await sqldb.queryAsync(sql.select_sync_job_sequences, {
-      course_id: res.locals.course.id,
-    });
-    res.locals.job_sequences = jobSequencesResult.rows;
+    const jobSequences = await queryRows(
+      sql.select_sync_job_sequences,
+      { course_id: res.locals.course.id },
+      JobSequenceRowSchema,
+    );
 
-    const result = await sqldb.queryAsync(sql.question_images, {
-      course_id: res.locals.course.id,
-    });
-    res.locals.images = result.rows;
-    res.locals.imageSyncNeeded = false;
+    const images = await queryRows(
+      sql.question_images,
+      { course_id: res.locals.course.id },
+      ImageRowSchema,
+    );
 
     if (config.cacheImageRegistry) {
       const ecr = new ECR(makeAwsClientConfig());
-      await async.each(res.locals.images, async (image) => {
-        var repository = new DockerName(image.image);
+      await async.each(images, async (image) => {
+        const repository = new DockerName(image.image);
         image.tag = repository.getTag() || 'latest (implied)';
         // Default to get overwritten later
         image.pushed_at = null;
         image.imageSyncNeeded = false;
         image.invalid = false;
 
+        /** @type {import('@aws-sdk/client-ecr').DescribeImagesCommandOutput} */
         let data;
         try {
-          data = await ecr.describeImages({
-            repositoryName: repository.getRepository(),
-          });
+          data = await ecr.describeImages({ repositoryName: repository.getRepository() });
         } catch (err) {
           if (err.name === 'InvalidParameterException') {
             image.invalid = true;
-            return null;
+            return;
           } else if (err.name === 'RepositoryNotFoundException') {
             image.imageSyncNeeded = true;
-            return null;
+            return;
           }
           throw err;
         }
 
-        const ecrInfo = {};
-        data.imageDetails?.forEach((imageDetails) => {
-          if (imageDetails.imageTags) {
-            imageDetails.imageTags.forEach((tag) => {
-              ecrInfo[imageDetails.repositoryName + ':' + tag] = imageDetails;
-            });
-          }
-        });
+        const repoName = repository.getCombined(true);
+        const ecrInfo = data.imageDetails?.find((imageDetails) =>
+          imageDetails.imageTags?.some(
+            (tag) => `${imageDetails.repositoryName}:${tag}` === repoName,
```

Comment:
I have no way to test this, but I considered using `tag === (repository.getTag() || 'latest')` for this comparison, assuming that the image details will presumably be limited to those of the given repository. Is this safe? I noticed that `repository.getCombined(true)` will also include a registry if it exists, would that be something to consider here? Anyway, to the best of my knowledge this is a cleaner way to replicate, as much as possible, the previously existing behaviour without changing this comparison.

Reply:
I think an even better option would be to use `imageIds` in the `describeImages` request above (https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_DescribeImages.html) to fetch information about only the specific tag we care about. Note that we'd then want to handle `ImageNotFoundException` and treat that as an `imageSyncNeeded = true` situation.

Reply:
Done in 74e31a891, but I have no way to test this.

Reply:
I tested this, continuing discussion here: https://github.com/PrairieLearn/PrairieLearn/pull/10073#discussion_r1683314739